spikeinterface_gui/view_base.py

```diff
@@ -9,47 +9,41 @@ class ViewBase():
     
     def __init__(self, controller=None, parent=None,  backend="qt"):
 
-
         self.backend = backend
         self.controller = controller
 
         if self.backend == "qt":
             # For QT the parent is the **widget**
-            from .backend_qt import SignalNotifyer, create_settings, listen_setting_changes
+            from .backend_qt import SignalNotifier, create_settings, listen_setting_changes
+            make_layout = self._qt_make_layout
             self.qt_widget = parent
-            self.notifyer = SignalNotifyer(parent=self.qt_widget)
-            if self._settings is not None:
-                create_settings(self, parent)
-            self._qt_make_layout()
-            if self._settings is not None:
-                listen_setting_changes(self, parent)
-
-
+            settings_kwargs = {"parent": parent}
         elif self.backend == "panel":
-            from .backend_panel import SignalNotifyer, create_settings, listen_setting_changes
-            self.notifyer = SignalNotifyer()
-            if self._settings is not None:
-                create_settings(self)
-            self._panel_make_layout()
-            if self._settings is not None:
-                listen_setting_changes(self)
-
+            from .backend_panel import SignalNotifier, create_settings, listen_setting_changes
+            make_layout = self._panel_make_layout
+            settings_kwargs = {}
 
+        self.notifier = SignalNotifier(view=self)
+        if self._settings is not None:
+            create_settings(self, **settings_kwargs)
+        make_layout()
+        if self._settings is not None:
+            listen_setting_changes(self)
```

Comment:
Lets keep this in if else, this is more easy to read and change.
I do not think we need such a grain of refactoring.
parent can be necessary somtimes on qt side.